app/workers/link_snp_phenotype.rb

```diff
@@ -0,0 +1,68 @@
+class LinkSnpPhenotype
+  include Sidekiq::Worker
+  sidekiq_options :queue => :link_snp_phenotype, :unique => true
+  attr_reader :snp, :characteristics, :papers_count
+
+  def perform(snp_id)
+    @snp = Snp.find(snp_id)
+    @characteristics = Phenotype.all.map { |x| x.characteristic }
+    @papers_count = @snp.snp_references.length
```

Comment:
`count` is probably better than `length` here. It sends a `COUNT(*)` query to the database. Unless `length` has been overwritten in `ActiveRecord`, which I'm not sure it is, `length` will lead to all the records being loaded just to count them.